eval/topdown_test.go

```diff
@@ -435,22 +545,82 @@ func TestTopDownEqExpr(t *testing.T) {
 
 }
 
-// TODO(tsandall): cover dereferencing of variables.
+func TestTopDownVarReferences(t *testing.T) {
+
+	tests := []struct {
+		note     string
+		rules    []string
+		expected interface{}
+	}{
+		// TODO(tsandall) dereferenced variables must be bound beforehand (safety check)
+		{"var ground", []string{"p[x] :- v = [[1,2],[2,3],[3,4]], x = v[2][1]"}, "[4]"},
+		{"var non-ground", []string{"p[x] :- v = [[1,2],[2,3],[3,4]], x = v[i][j]"}, "[1,2,2,3,3,4]"},
```

Comment:
sets typically don't have duplicates.  We should think about how we might address that.